dj_cqrs/delay.py

```diff
@@ -0,0 +1,63 @@
+#  Copyright © 2021 Ingram Micro Inc. All rights reserved.
+
+from datetime import timedelta
+from queue import PriorityQueue
+
+from dj_cqrs.utils import utc_now
+
+
+class DelayMessage:
+    """Delay message.
+
+    :param delivery_tag:
+    :type delivery_tag: int
+    :param payload:
+    :type payload: dj_cqrs.dataclasses.TransportPayload
+    """
+
+    def __init__(self, delivery_tag, payload, delay):
+        assert delay >= 0
+        self.delivery_tag = delivery_tag
+        self.payload = payload
+        self.eta = utc_now() + timedelta(seconds=delay)
+
+
+class DelayQueue:
+    """Queue for delay messages."""
+
+    def __init__(self):
+        self._queue = PriorityQueue()
```

Comment:
Just asking: in Python docs there is this: https://docs.python.org/3/library/queue.html#queue.PriorityQueue

```python
The lowest valued entries are retrieved first (the lowest valued entry is the one returned by sorted(list(entries))[0]). A typical pattern for entries is a tuple in the form: (priority_number, data).

If the data elements are not comparable, the data can be wrapped in a class that ignores the data item and only compares the priority number:

from dataclasses import dataclass, field
from typing import Any

@dataclass(order=True)
class PrioritizedItem:
    priority: int
    item: Any=field(compare=False)
```

Don't we need to wrap?

Reply:
No, we always have comparable eta timestamp for priority, custom put(delay_message) guaranties that.